game/tournament.py

```diff
@@ -36,22 +44,53 @@ def get_results_table(self):
     def get_scores(self):
         return self._scores_dict
 
+    def timeout_handler(self, arg1, arg2):
+        raise TimeLimitReached
+
     def run(self):
         for match in self._matches:
-            match.play_full_match()
+            abort_current_match = False
+            penalized_player = None
+            signal.signal(signal.SIGALRM, self.timeout_handler)
+            for player in match.get_players():
+                signal.setitimer(signal.ITIMER_REAL,self._prepare_time_limit)
```

Comment:
I was about to go with your suggested approach, but that would wait until the preparation is finished, no matter of how much it takes. The chosen approach interrupts the execution of the function after the timeout is reached. It's also the approach chosen by most of the guys on the forums, I'll add the necessary tests for it to ensure its reliability.